libraries/resource_providers/image_streamer/api300/deployment_plan.rb

```diff
@@ -0,0 +1,33 @@
+# (c) Copyright 2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+require_relative '../../../resource_provider'
+
+module OneviewCookbook
+  module ImageStreamer
+    module API300
+      # DeploymentPlan Provider resource methods
+      class DeploymentPlanProvider < ResourceProvider
+        def create_or_update
+          @item['goldenImageURI'] = load_resource(:GoldenImage, @context.golden_image)['uri']
+          @item['oeBuildPlanURI'] = load_resource(:BuildPlan, @context.build_plan)['uri']
+          super
+        end
+
+        def create_if_missing
+          @item['goldenImageURI'] = load_resource(:GoldenImage, @context.golden_image)['uri']
+          @item['oeBuildPlanURI'] = load_resource(:BuildPlan, @context.build_plan)['uri']
```

Comment:
Ah. I missed that. But it will still fail. `nil['uri']` doesn't work.

And I think it might be best to not have that method return nil if the name is not set. It should be up to the caller to call the method only if it should. If we can't use it here (because of the nil accessor error), I'm not sure where we'd be able to use it for that case.